erpnext/accounts/doctype/payment_entry/test_payment_entry.py

```diff
@@ -1301,8 +1304,55 @@ def test_receive_payment_from_payable_party_type(self):
 		)
 		self.voucher_no = pe.name
 		self.expected_gle = [
-			{"account": "_Test Cash - _TC", "debit": 1000.0, "credit": 0.0},
 			{"account": "Creditors - _TC", "debit": 0.0, "credit": 1000.0},
+			{"account": "_Test Cash - _TC", "debit": 1000.0, "credit": 0.0},
+		]
+		self.check_gl_entries()
+
+	def test_payment_against_partial_return_invoice(self):
+		"""
+		Checks GL entries generated for partial return invoice payments.
+		"""
+		si = create_sales_invoice(qty=10, rate=10, customer="_Test Customer")
+		credit_note = create_sales_invoice(
+			qty=-4, rate=10, customer="_Test Customer", is_return=1, return_against=si.name
+		)
+		pe = create_payment_entry(
+			party_type="Customer",
+			party="_Test Customer",
+			payment_type="Receive",
+			paid_from="Debtors - _TC",
+			paid_to="_Test Cash - _TC",
+		)
+		pe.set(
+			"references",
+			[
+				{
+					"reference_doctype": "Sales Invoice",
+					"reference_name": si.name,
+					"due_date": si.get("due_date"),
+					"total_amount": si.grand_total,
+					"outstanding_amount": si.outstanding_amount,
+					"allocated_amount": si.outstanding_amount,
+				},
+				{
+					"reference_doctype": "Sales Invoice",
+					"reference_name": credit_note.name,
+					"due_date": credit_note.get("due_date"),
+					"total_amount": credit_note.grand_total,
+					"outstanding_amount": credit_note.outstanding_amount,
+					"allocated_amount": credit_note.outstanding_amount,
+				},
+			],
+		)
+		pe.save()
+		pe.submit()
+		self.voucher_no = pe.name
+		self.expected_gle = [
```

Comment:
Not able to understand this expected GL Entry output, there is a Sales Invoice of amount 1000 (10*10) and a credit note of amount -40 (-4 * 10). So the net amount paid by the customer (entry in cash account) should be 960. Here the GL Entry is for 1000 which seems incorrect

What was the paid amount value in the PE here, can't see that being set either.

Reply:
Followed the same case as mentioned in the linked issue.

Created a SI with qty = 10, rate = 10, total = 100
Created a Return / Credit Note against the SI with qty = -4, rate= 10, total = -40
Created a PE with paid amount = 1000 (set through the default args of the method)

Allocated 100 against the SI and -40 against the Credit Note, Unallocated amount = 940 (in order to check if partial allocation entries also get posted correctly here)

100 gets credited against the SI so its outstanding becomes 0.
40 gets debited against the Credit Note so its outstanding becomes 0.
Rest 940 gets credited in Debtors against the paid amount 1000 through Cash.

Saw the GL entries generated for such a case before the changes to payment entry were made to test it here. I can fix the test if these are not the expected entries.

Reply:
The net value going into the cash account should be 960.